servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServiceTalkHttpAttributesGetter.java

```diff
@@ -117,35 +116,34 @@ private static final class ClientGetter extends ServiceTalkHttpAttributesGetter
         @Override
         @Nullable
         public String getUrlFull(final HttpRequestMetaData request) {
+            String requestTarget = request.requestTarget();
+            if (requestTarget.startsWith("https://") || requestTarget.startsWith("http://")) {
+                // request target is already absolute-form: just return it.
+                return requestTarget;
+            }
+
+            // in this case the request target is most likely origin-form so we need to convert it to absolute-form.
             HostAndPort effectiveHostAndPort = request.effectiveHostAndPort();
             if (effectiveHostAndPort == null) {
+                // we cant create the authority so we must just return.
                 return null;
             }
             String scheme = request.scheme();
             if (scheme == null) {
-                scheme = HTTP_SCHEME;
+                scheme = effectiveHostAndPort.port() == 443 ? HTTPS_SCHEME : HTTP_SCHEME;
```

Comment:
Just a note for anyone else looking at the PR: this is best effort. At client level, we don't know if the actually connection will be secure or not and therefore can not reliably detect scheme other than relying on standard port convention.

To make it work correctly, we may consider other options in the future. Some ideas are:
1. Exposing client configuration for filters
2. Using connection-level filter to attach `scheme` attribute to the current span after connection is selected